LookoutCASB: Initial version

Dhiraj3 opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [ ] Tests
- [ x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=13245) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=13245) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=13245) it.</sub>

marcellarichmond commented:
@mchasepan CipherCloud / Lookup pull request!

adi88d commented on Packs/LookoutCASB/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,142 @@
+[
+	{
+		"id": "CipherCloud Anomaly",
+		"version": 1,
```
```suggestion
		"version": -1,
```

Dhiraj3 replied:
fixed

adi88d commented on Packs/LookoutCASB/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,142 @@
+	},
+	{
+		"id": "CipherCloud Violation",
+		"version": 1,
```
```suggestion
		"version": -1,
```

Dhiraj3 replied:
fixed this and all version comments.

adi88d commented on Packs/LookoutCASB/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,142 @@
+	},
+	{
+		"id": "Dhiraj",
+		"version": 2,
```
```suggestion
		"version": -1,
```

adi88d commented on Packs/LookoutCASB/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,142 @@
+	},
+	{
+		"id": "Lookout CASB Anomaly",
+		"version": 2,
```
```suggestion
		"version": -1,
```

adi88d commented on Packs/LookoutCASB/IncidentTypes/customIncidentTypes.json:
```diff
@@ -0,0 +1,142 @@
+	},
+	{
+		"id": "Lookout CASB Violation",
+		"version": 2,
```
```suggestion
		"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+	"incidentFields": [
+		{
+			"id": "incident_actiontype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_activitytype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_anomalyname",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_anomalytype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_appname",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_cloudname",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_cloudtype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_contentname",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_currentcity",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_currenteventid",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_currenttimestamp",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_eventtype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_policyname",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_previouscity",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_previouseventid",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_previoustimestamp",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_scantype",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_timestamp",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_useremail",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d commented on Packs/LookoutCASB/IndicatorFields/incidentfields.json:
```diff
@@ -0,0 +1,887 @@
+		},
+		{
+			"id": "incident_violation",
+			"version": 1,
```
```suggestion
			"version": -1,
```

adi88d reviewed (changes requested):
@Dhiraj3 Thank you for your contribution! it looks great, few notes:
- Please split the files `incidentfields.json` and `customIncidentTypes.json`, each field and type should be in separate file.
- Update the description for end_time and start_time args with date format for example(i.e. ISO8601 as 2020-01-01T00:11:22Z).
- Add a readme file for the integration and for `Lookout-CASB-User-Risk-PlayBook` using [generate-docs](https://github.com/demisto/demisto-sdk#generate-docs) command.
- update `CASBIntegration_image.png` file - Logo should match the requirements (120x50 px PNG, transparent background, <10kB in Size).
- Add [unitests](https://xsoar.pan.dev/docs/integrations/unit-testing#write-your-unit-tests) to the integration.
- Your code must pass linters see [this](https://xsoar.pan.dev/docs/integrations/linting) article for more info and [lint](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/lint/README.md) command.

see my other comment below

Dhiraj3 commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,183 @@
+configuration:
+- display: Server URL (e.g. https://management-server.ciphercloud.com)
+  name: url
+  defaultvalue: https://krypton-ms.next.ciphercloud.in
```
fixed

Dhiraj3 commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,183 @@
+  required: false
+- display: Client Id
+  name: client_id
+  defaultvalue: cortexdev
```
fixed

Dhiraj3 commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,183 @@
+  required: true
+- display: Client Secret
+  name: client_secret
+  defaultvalue: XFrGiFHrQ8cG6a2ZMuYiKWBhf+4tLO7wg2aAdnyhDiA=
```
fixed

Dhiraj3 commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.py:
```diff
@@ -0,0 +1,718 @@
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client, event_type: str) -> str:
```
Yes this was used when testing.

Dhiraj3 commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,183 @@
+      description: 'Content info'
+      type: string
+    description: Get information about user or content.
+  - name: lookout-profile-user
```
fixed.

adi88d commented:
@Dhiraj3
- Update the description for end_time and start_time params with date format for example(i.e. ISO8601 as 2020-01-01T00:11:22Z).
- Add a readme file for the integration using [generate-docs](https://github.com/demisto/demisto-sdk#generate-docs) command.
- Add [unitests](https://xsoar.pan.dev/docs/integrations/unit-testing#write-your-unit-tests) to the integration.
- Renamae the incident fields name to starts with `incidentfield-` (for example: `incidentfields-LookoutCASB-actiontype.json` to `incidentfield-LookoutCASB-actiontype.json`)
- Please add readable_output to your commands (see this [page](https://xsoar.pan.dev/docs/reference/api/common-server-python#commandresults))

adi88d commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.py:
```diff
@@ -0,0 +1,746 @@
+                end_time=end_time,
+                max_results=max_results
+            )
+
```
readable_output =  tableToMarkdown('Violations', violations, removeNull=True, headerTransform=string_to_table_header)

adi88d commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.py:
```diff
@@ -0,0 +1,746 @@
+            cmd_result = CommandResults(
+                outputs_prefix='CipherCloud.Violation',
+                outputs_key_field=['eventId'],
+                outputs=violations
```
```suggestion
                outputs=violations,
                readable_output=readable_output
```

Dhiraj3 commented:
Addressed few review comments. Working on remain comments.

mchasepan commented on Packs/LookoutCASB/README.md:
```diff
@@ -0,0 +1,14 @@
+### Lookout CASB Integration
```
This content is a copy of the integration readme. Please delete this content and add a PACK readme (required) that talks about the value of the pack, what it does, what Lookout does, links to your site, etc. Let us know if you need help or example.

Dhiraj3 replied:
Sure will change it.

adi88d replied:
please update this file

adi88d commented:
@Dhiraj3 Did you finished working on my comments?

Dhiraj3 commented:
Fixed most of the comments. But still need to address few comments, and validate.

> On Jul 13, 2021, at 7:24 AM, Adi Daud ***@***.***> wrote:
> 
> ﻿
> @Dhiraj3 Did you finished working on my comments?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

adi88d commented:
@Dhiraj3 Did you finished working on my comments?

Dhiraj3 commented:
Cloud you please re-review - I have addressed all the comments except for additional tests and rename file.

> On Jul 19, 2021, at 10:40 PM, Adi Daud ***@***.***> wrote:
> 
> 
> @Dhiraj3 <https://github.com/Dhiraj3> Did you finished working on my comments?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/demisto/content/pull/13245#issuecomment-883078332>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABNOMCNCESR5N2EAH5243DTTYUD37ANCNFSM46YDMPGA>.
>

adi88d commented on Packs/LookoutCASB/Playbooks/Lookout-CASB-User-Risk-PlayBook_README.md:
```diff
@@ -0,0 +1,29 @@
+This playbook does not use any sub-playbooks.
+
+### Integrations
+This playbook does not use any integrations.
```
```suggestion
* CASBIntegration
```

adi88d commented on Packs/LookoutCASB/Playbooks/Lookout-CASB-User-Risk-PlayBook_README.md:
```diff
@@ -0,0 +1,29 @@
+
+## Playbook Image
+---
+![Lookout-CASB-User-Risk-PlayBook](Insert the link to your image here)
```
```suggestion
![Lookout-CASB-User-Risk-PlayBook](../../doc_files/Lookout-CASB-User-Risk.png)
```

adi88d commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,187 @@
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: End Time
```
```suggestion
- display: End Time(in ISO8601 format)
```

Dhiraj3 replied:
updated

adi88d commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,187 @@
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Start Time
```
```suggestion
- display: Start Time(in ISO8601 format)
```

Dhiraj3 replied:
updated

adi88d commented on Packs/LookoutCASB/Integrations/CASBIntegration/CASBIntegration.yml:
```diff
@@ -0,0 +1,187 @@
+  type: 13
+  required: false
+- display: Maximum number of incidents per fetch
+  name: max_fetch
```
```suggestion
  name: max_fetch
```

this variable only configured in the yml file but you dont use it at all, please limit the fetched incidents with this value

Dhiraj3 replied:
max_fetch is optional. There is a max limit at the server side. This field is provided for client as optional if client would like to limit the records.

adi88d reviewed (commented):
@Dhiraj3 few more notes:

- Add unitests to the integration.
- Add a readme file for the integration using [generate-docs](https://github.com/demisto/demisto-sdk#generate-docs) command.
- Add the command `lookout-add-user`
- Add limit argument to the commands:
   - lookout-get-anomalies
   - lookout-get-events
   - lookout-get-information
   - lookout-get-violations
   
see my other comment below

Dhiraj3 commented:
Our use case does not need command `lookout-add-user`

adi88d commented:
@Dhiraj3 please fix my last comments:

- Add unitests to the integration.
- Add a readme file for the integration using [generate-docs](https://github.com/demisto/demisto-sdk#generate-docs) command.
- Add limit argument to the commands:
   - lookout-get-anomalies
   - lookout-get-events
   - lookout-get-information
   - lookout-get-violations
- Please use the value from `max_fetch` parameter inside the python file to limit the fetched incidents.

ShahafBenYakir commented:
Closing as we didn't hear back from the partner.
